### PR TITLE
#96 Updated openapi file path for documentation deployment

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Fixes #
Updated redocly commands for the new openApi file path. 
there should be no further changes necessary. but not sure until it merges with the master. cause the deploy doc only runs on master.

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [ ] New feature
- [x] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: .github/workflows/deploy-docs.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       - name: Install Redoc
         run: npm install -g @redocly/cli@latest
       - name: Combine OpenAPI specs
-        run: redocly bundle docs/openapi/v1/api.yaml -o docs/openapi/v1/combined.yaml
+        run: redocly bundle openapi/versions/v1.yml -o docs/openapi_combined.yaml
```
Comment (melon-ruet) [inline .github/workflows/deploy-docs.yml:37]: If redocly fails to create proper combined api speces, we will not use it

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -25,7 +25,7 @@ DOCKER_DEV_CONTAINER_NAME?=bloodconnect-dev
 
 # Documentation
 sphinx-html:
-	redocly bundle docs/openapi/v1/api.yaml -o docs/openapi/v1/combined.yaml
+	redocly bundle openapi/versions/v1.yml -o docs/openapi_combined.yaml
```
Comment (melon-ruet) [inline Makefile:28]: If you change the path, you need to fix it in docs/conf.py

```
redoc = [{
    "name": "BloodConnect API",
    "page": "openapi/api",
    "spec": f"openapi/v1/combined.yaml"
}]
```

Comment (melon-ruet) [inline Makefile:28]: Please use docs/openapi/v1.yaml

File: docs/conf.py [hunk 3]
```diff
@@ -39,5 +39,5 @@
 redoc = [{
     "name": "BloodConnect API",
     "page": "openapi/api",
-    "spec": f"openapi/v1/combined.yaml"
+    "spec": f"openapi_combined.yaml"
```
Comment (melon-ruet) [inline docs/conf.py:42]: openapi/v1.yaml

File: .github/workflows/deploy-docs.yml (hunk 1 above)
Comment (melon-ruet) [inline .github/workflows/deploy-docs.yml:37]: Change docs/openapi_combined.yaml to docs/openapi/v1.yaml

File: Makefile (hunk 2 above)
Comment (mahinuzzaman) [reply Makefile:28]: ok vai

File: .github/workflows/deploy-docs.yml
Comment (melon-ruet) [inline .github/workflows/deploy-docs.yml:1]: We need to deploy docs on openapi change. Add openapi path here

```
    paths:
      - docs/**
```

**Review by melon-ruet — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-docs.yml [hunk 4]
```diff
@@ -34,7 +34,7 @@ jobs:
       - name: Install Redoc
         run: npm install -g @redocly/cli@latest
       - name: Combine OpenAPI specs
-        run: redocly bundle docs/openapi/v1/api.yaml -o docs/openapi/v1/combined.yaml
+        run: make bundle-openapi
```
[the remark this answers is not in the document]
Comment (mahinuzzaman) [reply .github/workflows/deploy-docs.yml:37]: oh. ok I'm working on it.

**Pull request closed without merging**